File: keiner_lindarte_aro/instructor_henry/mis_programas/dart/orientado_a_objetos/5_clase_de_09_agosto/publico_privado.dart
class Carro{
  // los atributos por defecto son publicos
  String? marca;
  String? color;
  // los atributos con un underscore al principio son privados
  // ignore: unused_field
  int? _precio = 1000000;

  Carro({this.marca,this.color});

  String informacion(){
    return "El carro comprado es de la marca ${marca}, con el color ${color}.";
  }
}

void main (){

  Carro carro = new Carro(color: "verde",marca: "toyota");
  print(carro.informacion());
  
}

File: keiner_lindarte_aro/instructor_henry/mis_programas/ruby/orientado_a_objetos/5_clase_de_09_agosto/publico_privado.rb
class Carro
  # por defecto son publicos
  attr_accessor :marca, :color, :precio
  # con esto ponemos para que los atributos o metodos sean privados
  private :precio

  precio = 1000000

  def initialize(marca,color)
    self.marca = marca
    self.color = color
  end

  def informacion
    puts "El carro comprado es de la marca #{self.marca}, con el color #{self.color}."
  end

end

carro = Carro.new("toyota","verde")
carro.informacion